### PR TITLE
Community feedback build

### PR DESCRIPTION
**Pull request by seanturner83**

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,2 @@
+#!groovy
```
Comment (oleg-nenashev) [inline Jenkinsfile]: Not required for common IDEs and GitHub

File: pom.xml [hunk 2]
```diff
@@ -10,7 +10,7 @@
     <artifactId>artifactdeployer</artifactId>
     <packaging>hpi</packaging>
     <name>Jenkins Artifact Deployer Plug-in</name>
-    <version>0.40</version>
+    <version>1.00</version>
```
Comment (oleg-nenashev) [inline pom.xml]: better to replace by `1.0-SNAPSHOT` and ensure that the common release flow from Parent POM is used

File: src/main/java/org/jenkinsci/plugins/artifactdeployer/ArtifactDeployerEntry.java [hunk 3]
```diff
@@ -46,7 +46,7 @@
     private boolean flatten;
     private boolean deleteRemote;
     private boolean deleteRemoteArtifacts;
-    private boolean deleteRemoteArtifactsByScript=false;
+    private transient boolean deleteRemoteArtifactsByScript=false;
```
Comment (oleg-nenashev) [inline src/main/java/org/jenkinsci/plugins/artifactdeployer/ArtifactDeployerEntry.java]: not sure what is this change, no clariication in the PR

File: Jenkinsfile (hunk 1 above)
Comment (seanturner83) [inline Jenkinsfile]: Ta :)

File: pom.xml (hunk 2 above)
Comment (seanturner83) [inline pom.xml]: Honestly, I don't know what this actually means.. researching

File: src/main/java/org/jenkinsci/plugins/artifactdeployer/ArtifactDeployerEntry.java (hunk 3 above)
Comment (seanturner83) [inline src/main/java/org/jenkinsci/plugins/artifactdeployer/ArtifactDeployerEntry.java]: Suggested by @daniel-beck, although I may have interpreted it wrongly.

Comment (daniel-beck) [inline src/main/java/org/jenkinsci/plugins/artifactdeployer/ArtifactDeployerEntry.java]: The idea is that this doesn't get stored to disk anymore as the related feature is gone. This depends on https://github.com/jenkinsci/artifactdeployer-plugin/compare/artifactdeployer-0.33...master

File: pom.xml [hunk 4]
```diff
@@ -84,4 +91,16 @@
...
         </pluginRepository>
     </pluginRepositories>
+<build>
+    <plugins>
+      <plugin>
+        <groupId>org.jenkins-ci.tools</groupId>
+        <artifactId>maven-hpi-plugin</artifactId>
+        <extensions>true</extensions>
+        <configuration>
+          <compatibleSinceVersion>1.0</compatibleSinceVersion>
```
Comment (daniel-beck) [inline pom.xml]: Well, 0.40 really, no?

Comment (seanturner83) [inline pom.xml]: not getting released :)

Comment (daniel-beck) [inline pom.xml]: https://github.com/jenkinsci/artifactdeployer-plugin/releases ?

Comment (seanturner83) [inline pom.xml]: lol, oops

Comment (seanturner83) [inline pom.xml]: updated to identify as a pre-release

**Pull request merged**